src/js/server/middleware.js

```diff
@@ -0,0 +1,92 @@
+// "Gatekeeper" middleware that rejects any incoming payloads that are not valid according to the schema provided in
+// `options.schemaContent`.  You can either pass the full content of the schema directly to `options.schemaContent` or
+// indicate the name of a schema file by setting `options.schemaFile` to the full path to your schema file.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./handler");
+
+fluid.defaults("gpii.schema.middleware.handler", {
+    gradeNames: ["gpii.express.handler"],
+    invokers: {
+        handleRequest: {
+            func: "{that}.sendResponse",
+            args: ["{that}.options.statusCode", "{that}.options.message"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.schema.middleware");
+
+gpii.schema.middleware.getMiddleware = function (that) {
+    return that.rejectOrForward;
+};
+
+gpii.schema.middleware.rejectOrForward  = function (that, req, res, next) {
+    if (that.options.schemaContent) {
+        var results = that.validator.validate("schema", req.body);
+        if (results) {
+            // Instantiate a handler that will take care of the rest of the request.
+            that.events.onInvalidRequest.fire(req, res, 500, results);
+        }
+    }
+    else {
+        // We choose to fail if we can't validate the body.  That way anything downstream is guaranteed to
+        // only receive 2valid content.
+        var message = "Your gpii.schema.middleware instance doesn't have a schema to work with, so it can't validate anything.";
+        that.events.onInvalidRequest.fire(req, res, 500, { ok: false, error: message });
+        fluid.fail(message);
+    }
+
+    next();
+};
+
+// TODO:  Extract key pieces from `gpii.express.requestAware.router` to handle the request with the common infrastructure
+
+fluid.defaults("gpii.schema.middleware", {
+    gradeNames: ["gpii.express.middleware"],
+    path: "/",
```

Comment:
Puzzling - we need to review how this grade structure works. It seems unusual to specify a "path" field in a thing called "middleware" - which, to me, should really only be honoured by a thing called a "router" - isn't it the job of middleware to not in general be aware what path it is bound to?